### PR TITLE
Added ability to use local swagger.json files in layout-swagger without absolute url

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**

### CODE REVIEW

**Comment by Blackbaud-SteveBrush** [comment]
@blackbaud-brandonhare: Got any steps for how I'd test this?

**Comment by blackbaud-brandonhare** [comment]
Hey @Blackbaud-SteveBrush , 

So to test this, all you need to do is have a test swagger spec in the data folder. Then, in the md file where you want to display the swagger-ui, put the following in the YFM:

```
swagger:
    input: name-of-json-file
    inputType: json
```

File: src/partials/partial-swagger.hbs [hunk 1]
```diff
@@ -1,7 +1,12 @@
 <div swagger-ui
-  input-type="url"
+  input-type="{{ swagger.inputType }}"
+  {{# compare swagger.inputType '===' 'json' }}
+  input="{{ getDataByName swagger.input }}"
+  {{/ compare }}
+  {{# compare swagger.inputType '===' 'url' }}
```
Comment (Blackbaud-SteveBrush) [inline src/partials/partial-swagger.hbs:7]: How many input types are there? Seems that this wouldn't work if I didn't include either `json` or `url` as the type? If the default is `url`, then perhaps we could do a catch-all helper here:

```
<div swagger-ui
  input-type="{{ swagger.inputType }}"
  {{# compare swagger.inputType '===' 'json' }}
    input="{{ getDataByName swagger.input }}"
  {{ else }}
    input="'{{ swagger.input }}'"
  {{/ compare }}
  {{# if swagger.template }}
    template-url="{{ swagger.template }}"
  {{/ if }}
></div>
```

If there ARE more options than `json` and `url`, what you have is okay; some folks' implementations just might not work if they add an incorrect `inputType` (which may be acceptable for this).

**Comment by Blackbaud-SteveBrush** [comment]
Overall, this is awesome. Getting swagger implemented for Stache is a huge win. Let me know what you think about my comment above.

**Comment by blackbaud-brandonhare** [comment]
So there is one more type, YAML. We don't currently support it, but we could if there was demand for it. The YAML requires a custom parser. But I don't see anyone wanting to use YAML over json honestly. I think once people see how the json implementation works, they'll want to use that one instead of the URL. For me, json = files stored in Stache, url = files stored outside of Stache. So maybe we should default to json and then have the option to override if they want to use a URL?

**Comment by Blackbaud-SteveBrush** [comment]
Defaulting to using the **/data** folder is a good call. I'd go ahead and represent all three input types in the partial, too, if it's not too difficult.

**Comment by blackbaud-brandonhare** [comment]
@Blackbaud-SteveBrush Hmm, I tested with GLX, and the input-type was blank when I omitted it.

File: src/partials/partial-swagger.hbs [hunk 2]
```diff
@@ -1,5 +1,9 @@
 <div swagger-ui
-  input-type="{{ swagger.inputType }}"
+  {{# if swagger.inputType }}
```
Comment (blackbaud-brandonhare) [inline src/partials/partial-swagger.hbs:2]: What about just doing this @Blackbaud-SteveBrush ?

**Comment by Blackbaud-SteveBrush** [comment]
I can't replicate on my end. Which page? What does your FM look like?

**Comment by blackbaud-brandonhare** [comment]
I just removed the inputType variable from one of the GLX pages I was testing with after updating Stache. It didn't throw any errors, but the inputType was empty. 

```
swagger:
    input: givalanche-public-data-service
```

**Comment by Blackbaud-SteveBrush** [comment]
Ah, I see it now.

**Comment by Blackbaud-SteveBrush** [comment]
Hm. Still can't reproduce the problem. Seems to work okay over here. 

Here's my FM:

```
---
layout: layout-swagger
swagger:
    input: givalanche-email-processing-service
---
```

And here's what I see when I output the `swagger` config to the page:

```
{"inputType":"json","input":"givalanche-email-processing-service"}
```

And here's my HTML:
screen shot 2016-06-16 at 4 32 26 pm

**Comment by Blackbaud-SteveBrush** [comment]
@blackbaud-brandonhare Actually, I'm okay with your update to the partial. If everything else works as expected, we can publish.

**Comment by blackbaud-brandonhare** [comment]
Approved.

[Approved with PullApprove](https://pullapprove.com/blackbaud/stache/pull-request/77/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=blackbaud/stache)

**Comment by Blackbaud-SteveBrush** [comment]
Approved!

[Approved with PullApprove](https://pullapprove.com/blackbaud/stache/pull-request/84/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=blackbaud/stache)

**Pull request merged**